### DISCUSSION TITLE
Interpolation and Clipping

### DISCUSSION THREAD

**Discussion by GmBinom** [ImageMagick/ImageMagick#4757] [Help] [answered]
Hi everyone,

i am using clipping pathes to remove the background of an image.
sadly I get very edgy results, because ImageMagick doesn't interpolate during the clipping.

In Photoshop you would usually use soft selections of 0.5px to hide this situation. But even without this, the clipping is really harsh on the decision of leave the pixel there or delete it. its kind of binary deciding instead of "interpolating"

`convert basefile.tif -alpha transparent -clip -alpha opaque +clip -trim +repage resultfile.tif`
![im_clipping](https://user-images.githubusercontent.com/98460728/151155026-1d80db75-a6b4-4bf7-a949-99ddbde7370f.png)

Here the behavior of Photoshop
![ps_clipping](https://user-images.githubusercontent.com/98460728/151155126-7a4f8823-6dc9-4386-8041-1836e778d209.png)

Has anyone an idea how to get equal results? The goal is a result like Photoshop with soft selection like this:
![ps_clipping_soft_selection](https://user-images.githubusercontent.com/98460728/151155511-0dda5e45-b427-4b66-bf8b-fb943cd5870f.png)

Thanks in advance for your thoughts and help.
You are awesome!

**Comment by snibgo** [comment]
Yes, clipping is binary. Softer edges can be had by rasterizing the clipping path to get anti-aliased edges, then copy that to the alpha channel. You can also blur edges if you want.

If you need more help, please state the version of IM and platform, and give a sample input file.

**Reply by GmBinom** [reply]
This sounds great, but I honestly don't know how I would achieve that with ImageMagick

```
Version: ImageMagick 6.9.9-40 Q16 x86_64 2019-02-21 http://www.imagemagick.org
Copyright: © 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Features: Cipher DPC Modules 
Delegates (built-in): bzlib djvu fftw fontconfig freetype gslib jbig jng jp2 jpeg lcms ltdl lzma openexr png ps raw tiff webp x xml zlib
```

[sample.jpg.zip](https://github.com/ImageMagick/ImageMagick/files/7941559/sample.jpg.zip)

**Reply by GmBinom** [reply]
i am using macOS 10.15.7 right now

**Comment by snibgo** [comment, marked as answer]
With IM v6, on bash:
```
identify -quiet -format "%[8BIM:1999,2998:#1]" sample.jpg >temppath.svg

convert sample.jpg \( temppath.svg -negate +write p.png \) -alpha off -compose CopyOpacity -composite -trim +repage x.png
```
x.png is the result. If you want to add a blur:

```
convert sample.jpg \( temppath.svg -negate -blur 0x0.5 \) -alpha off -compose CopyOpacity -composite -trim +repage x2.png
```
temppath.svg is a temporary file, and can be deleted.

**Reply by GmBinom** [reply]
Thanks, i ll keep you informed, if it fits for our usecase

**Reply by GmBinom** [reply]
It needed some adjustments for TIFF Files, but I got acceptable results :-)

Sadly Photoshop is still way better, but I can accept that.
